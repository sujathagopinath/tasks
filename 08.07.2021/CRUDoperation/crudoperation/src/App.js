import './App.css';
import { Switch, Route } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';
import "react-toastify/dist/ReactToastify.css";
import Crud from './crud';
import Update from './crud/update';
function App() {
  return (
    <div className="App">
      {/* <h2>CRUD operation</h2> */}
      <ToastContainer />
      <Switch>
        <Route exact path="/" component={Crud} />
        <Route exact path="/update/:id" component={Update} />
      </Switch>
    </div>
  );
}

export default App;
