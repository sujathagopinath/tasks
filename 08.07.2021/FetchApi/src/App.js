import React from 'react';

function App(){
    return(
        <div>
            <FetchApi/>
        </div>
    )
}

export default App;