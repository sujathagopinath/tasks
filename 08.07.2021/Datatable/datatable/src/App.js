import React from 'react'
import logo from './logo.svg';
import './App.css';

import Home from './home';
import { EmployeeProvider } from './context'
function App() {
  return (
    <div className="App">
      <EmployeeProvider>

        <Home />
      </EmployeeProvider>

    </div>
  );
}

export default App;

