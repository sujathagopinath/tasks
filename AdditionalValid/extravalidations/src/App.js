import './App.css';
import { toast } from 'react-toastify'
import 'react-toastify/dist/ReactToastify.css';
import Modal from 'react-modal'

toast.configure()

function App() {
  const Notify = () => {
    toast('Basic Notify', { position: toast.POSITION.TOP_LEFT })
  toast.success('Completed',{position:toast.POSITION.TOP_RIGHT})     
  }
  
  return (
    <div className="App">
      <h5>Toastify</h5>
      <button onClick={Notify}>Notify</button>
      <h4>Modal</h4>

      <Modal isOpen={true}>
        <h2>Modal title</h2>
      </Modal>
    </div>
  );
}

export default App;
