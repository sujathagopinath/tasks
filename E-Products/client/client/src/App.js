import './App.css';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import PrivateRoute from './utils/PrivateRoute';
import role from './Constants/Roles';
import Header from './components/Header/Header';
import Home from './components/Home/Home';
import SignUp from './components/Signup/signup';
import SignIn from './components/Signin/signin';
import Profile from './components/Userprofile/Profile';
import ListUsers from './components/Userprofile/ListUsers';
import UpdateProfile from './components/Updateuser/UpdateProfile';
import AddProduct from './components/Products/AddProduct';
import ProductUpdate from './components/Products/ProductUpdate';
import Products from './components/Products/Products';
import Checkout from './components/Products/Checkout'
import Forbidden from './Constants/Forbidden';
import Cart from './components/Products/cart';

function App() {
  return (
     <>
      <BrowserRouter>
        <Header />
        <Routes>
          <Route exact path="/" element={<Home />}  />
          <Route path='/signup' element={<SignUp />} />
          <Route path='/signin' element={<SignIn />} />
          <Route path='/forbidden' element={<Forbidden />} />
        
          <Route path='/products'  element=
            {<PrivateRoute > <Profile role={role.admin} /> </PrivateRoute>} />
          <Route path='/allusers' element=
            {<PrivateRoute>  <ListUsers role={role.admin} /> </PrivateRoute>}  />  
          <Route path='/create' element=
            {<PrivateRoute> <AddProduct role={role.admin}/> </PrivateRoute>}  /> 
          <Route path='/updateproduct' element=
            {<PrivateRoute> <ProductUpdate role={role.admin} /> </PrivateRoute>} />
          
          <Route path='/update' element=
            {<PrivateRoute> <UpdateProfile role={role.user} />  </PrivateRoute>} /> 
         <Route path='/allproducts' element=
            {<PrivateRoute> <Products role={role.user} /> </PrivateRoute>} /> 
          <Route path='/cart' element=
             {<PrivateRoute> <Cart role={role.user} />  </PrivateRoute>} /> 
          <Route path='/checkout' element=
            {<PrivateRoute> <Checkout role={role.user}/> </PrivateRoute>}  />  
          
          
      </Routes>
        </BrowserRouter>
    </>   
  );
}

export default App;
