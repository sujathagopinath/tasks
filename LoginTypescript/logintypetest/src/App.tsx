import React from "react";
import "./App.css";
import { Login } from "./login";
import Users from "./users";

function App() {
  return (
    <div className="App">
      <Login />
      <Users />
    </div>
  );
}

export default App;
