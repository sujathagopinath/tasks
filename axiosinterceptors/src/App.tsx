import React from "react";
import "./App.css";
import Customaxios from "./axios";
import { Route, Routes, BrowserRouter } from "react-router-dom";

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Customaxios fetchData />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
