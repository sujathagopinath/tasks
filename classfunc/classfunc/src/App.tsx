import React from "react";
import "./App.css";
import Step1 from "./components/userPersonal";

function App() {
  return (
    <div className="App">
      <Step1 />
    </div>
  );
}

export default App;
