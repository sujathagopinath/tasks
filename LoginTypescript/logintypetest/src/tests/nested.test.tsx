import * as React from "react";
import { shallow, mount } from "enzyme";
import App from "../App";
import toJson from "enzyme-to-json";
import Parentform from "../components/parent";
import Childform from "../components/child";

describe("render parent", () => {
  const wrapper = shallow(<Parentform />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
describe("Parent Component", () => {
  it("renders Child component", () => {
    const wrapper = shallow(<Parentform />);
    wrapper.setProps({
      state: "test",
      placeholder: "some letters",
    });
    expect(wrapper.containsMatchingElement(<Childform />)).toEqual(true);
  });
});

describe("Parent render", () => {
  jest.mock("../components/child", () => "Childform");
  it("should render the parent", () => {
    //       const props = {
    //           state: "hello",
    //   setState: (val: string)=>"world",
    //   handleOnSubmit: (e: SubmitEvent) =>void,
    //   placeholder: "tyoe some letters"
    //       };
    const tree = mount(
      <Childform>
        <Parentform />
      </Childform>
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
