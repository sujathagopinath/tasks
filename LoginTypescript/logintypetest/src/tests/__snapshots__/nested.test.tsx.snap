// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Childform
  data-testid="output"
  handleOnSubmit={[Function]}
  placeholder="Type some letters"
  setState={[Function]}
  state=""
/>
`;
